File: OurKnowledgeBackend/src/main/java/com/example/ourknowledgebackend/service/TechnologyService.java
package com.example.ourknowledgebackend.service;

import com.example.ourknowledgebackend.exceptions.*;
import com.example.ourknowledgebackend.model.TechnologyTree;

import java.util.List;

public interface TechnologyService {

    List<TechnologyTree> listRelevantTechnologies();

    List<TechnologyTree> addTechnology( Long userId, String name, Long parentId, boolean relevant) throws InstanceNotFoundException, DuplicateInstanceException, PermissionException;

    List<TechnologyTree> deleteTechnology(Long userId, Long technologyId, boolean deleteChildren) throws HaveChildrenException, PermissionException, InstanceNotFoundException;
}

File: our-knowledge-front/src/backend/technologyService.js
import {config, appFetch} from './appFetch';

export const listRelevantTechnologies = (onSuccess) =>
    appFetch(`/technology/list`, config('GET'), onSuccess);

export const addTechnology = (name, parentId,
    onSuccess, onErrors) =>
    appFetch(`/technology/add`,
        config('POST', {name, parentId}), onSuccess, onErrors);

export const deleteTechnology = (technologyId, deleteChildren,
    onSuccess, onErrors) =>
    appFetch(`/technology/delete`,
        config('POST', {technologyId, deleteChildren}), onSuccess, onErrors);
